### PR TITLE
#153249007  Fix Flask  Endpoints.

### PR DESCRIPTION
**Pull request by betsybeth**
##### has the following:
* views.py

### CODE REVIEW

**Comment by meshnesh** [comment]
* Good implementation of error handling
* Good implementation of password validation, although you could make it a bit less cluttered

**Comment by Rodgers-M** [comment]
Great fix for  the password checks and the removal of white spaces in input

File: app/views.py [hunk 1]
```diff
@@ -20,18 +20,26 @@ def not_found(error):
...
+    })
     response.status_code = 500
     return response
+@app.errorhandler(405)
+def not_allowed(error):
+    """Handle 405 error."""
+    response = jsonify({"message": "Sorry the method is not allowed"})
+    response.status_code = 405
+    return response
+
```
Comment (ann-mukundi) [inline app/views.py:36]: PEP 8, too many blank lines.

File: app/views.py [hunk 2]
```diff
@@ -197,13 +208,14 @@ def delete_event(eventId):
...
+                user.delete_event(eventId)
+                message = {" message ": "event deleted"}
+                response = jsonify(message)
+                response.status_code = 204
+                return response
+            return jsonify({"message": "not found, event already deleted"})
+    return jsonify(response)
 
 
 @app.route("/api/v1/event/<eventId>/rsvp", methods=["POST"])
```
Comment (ann-mukundi) [inline app/views.py:221]: When you move to challenge 3 try not to repeat sections like <b> /api/v1 </b> over an over. Look up DRY( Do not Repeat Yourself) Principle

File: app/views.py [hunk 3]
```diff
@@ -105,31 +116,32 @@ def create_event():
...
-                if not isinstance(date, str):
-                    message = {"message": "invalid date"}
-                if author.strip() == "" or not author.isalpha():
-                    message = {"message": "invalid author"}
-                    return jsonify(message)
-                if location.strip() == "" or not location.isalpha():
-                    message = {"message": "invalid location"}
-                    return jsonify(message)
-                user.create_event(name, description, category, date, author,
+    for user in users:
```
Comment (ann-mukundi) [inline app/views.py]: Why are we looping users when creating an event?

**Review by ann-mukundi — COMMENTED** [review]
Great work Beth.
In addition to the PR reviews here are a few more reviews that were not requested in this PR.
- Event author can be the name of the person who is logged in when creating an event instead of the user inputing it
- Please catch errors that occur when a field is missing when adding an event
- Error messages such as invalid description does not tell me what is wrong with my description
- How comes event name, description and location don’t allow spaces for example <b> Andela offices </b> returns an invalid location error
- I can’t add more that one event
- I believe an event can be deleted directly using the id so as to avoid looping the users first.

**Pull request merged**